#!/usr/bin/env python
PACKAGE = "worldModel"

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
   
gen.add("filterXYownTolerance", double_t, 0, "A double parameter", 0.0, 0.0, 3.0)
gen.add("filterXYmemberTolerance", double_t, 0, "A double parameter", 0.0, 0.0, 3.0)
gen.add("trackerXYTolerance", double_t, 0, "A double parameter", 0.0, 0.0, 10.0)
gen.add("trackerTimeout", double_t, 0, "A double parameter", 0.0, 0.0, 10.0)
gen.add("extrapolationTimeout", double_t, 0, "A double parameter", 0.0, 0.0, 10.0)
gen.add("discriminatorLimitX", double_t, 0, "A double parameter", 0.0, 0.0, 10.0)
gen.add("discriminatorLimitY", double_t, 0, "A double parameter", 0.0, 0.0, 10.0)
gen.add("outlierNSigma", double_t, 0, "A double parameter", 0.0, 3.0, 6.0)
gen.add("outlierIterFraction", double_t, 0, "A double parameter", 0.1, 0.0, 1.0)
gen.add("outlierMaxIter", int_t, 0, "A Integer parameter", 1, 1, 5)
gen.add("speedFitOrder", int_t, 0, "A Integer parameter", 0, 0, 3)
gen.add("speedMinSize", double_t, 0, "A double parameter", 0.3, 0.0, 1.0)
gen.add("speedMaxSize", double_t, 0, "A double parameter", 5.0, 0.0, 10.0)
gen.add("speedResidualThreshold", double_t, 0, "A double parameter", 1.0, 0.0, 5.0)

exit(gen.generate(PACKAGE, "ObstacleTracker", "obstacletracker"))
